src/mudlet-lua/tests/GUIUtils.lua

```diff
@@ -0,0 +1,135 @@
+  end)
+
+  describe("Test the operation of the ansi2decho function", function()
+
+    it("Should have loaded the function successfully", function()
+      assert.truthy(ansi2decho)
+    end)
+
+    it("Should convert simple single ANSI sequences correctly", function()
+      local sequences = {
+        {"\27[0m", "<r>"},
+        {"\27[30m", "<0,0,0:>"},
+        {"\27[31m", "<128,0,0:>"},
+        {"\27[32m", "<0,179,0:>"},
+        {"\27[33m", "<128,128,0:>"},
+        {"\27[34m", "<0,0,128:>"},
+        {"\27[35m", "<128,0,128:>"},
+        {"\27[36m", "<0,128,128:>"},
+        {"\27[37m", "<192,192,192:>"},
+        {"\27[40m", "<:0,0,0>"},
+        {"\27[41m", "<:128,0,0>"},
+        {"\27[42m", "<:0,179,0>"},
+        {"\27[43m", "<:128,128,0>"},
+        {"\27[44m", "<:0,0,128>"},
+        {"\27[45m", "<:128,0,128>"},
+        {"\27[46m", "<:0,128,128>"},
+        {"\27[47m", "<:192,192,192>"},
+        {"\27[90m", "<128,128,128:>"},
+        {"\27[91m", "<255,0,0:>"},
+        {"\27[92m", "<0,255,0:>"},
+        {"\27[93m", "<255,255,0:>"},
+        {"\27[94m", "<0,0,255:>"},
+        {"\27[95m", "<255,0,255:>"},
+        {"\27[96m", "<0,255,255:>"},
+        {"\27[97m", "<255,255,255:>"},
+        {"\27[100m", "<:128,128,128>"},
+        {"\27[101m", "<:255,0,0>"},
+        {"\27[102m", "<:0,255,0>"},
+        {"\27[103m", "<:255,255,0>"},
+        {"\27[104m", "<:0,0,255>"},
+        {"\27[105m", "<:255,0,255>"},
+        {"\27[106m", "<:0,255,255>"},
+        {"\27[107m", "<:255,255,255>"},
+      }
+      for _, seq in ipairs(sequences) do
+        local actualResult = ansi2decho(seq[1])
+        assert.are.same(seq[2], actualResult)
+      end
+    end)
+
+    it("Should combine tags correctly", function()
+      local sequences = {
+        {"\27[0;30m", "<r><0,0,0:>"},
+        {"\27[1;30m", "<128,128,128:>"},
+        {"\27[1;40m", "<:0,0,0>"},
+        {"\27[31;42m", "<128,0,0:0,179,0>"},
+        {"\27[30;0m", "<r>"},
+        {"\27[0;1;30;40m", "<r><128,128,128:0,0,0>"},
+      }
+      for _, seq in ipairs(sequences) do
+        local actualResult = ansi2decho(seq[1])
+        assert.are.same(seq[2], actualResult)
+      end
+    end)
+
+    it("Should leave normal text and other escape sequences alone", function()
+      local sequences = {
+        {"Hello World", "Hello World"},
+        {"[Something in braces]", "[Something in braces]"},
+        {"\27[4z<PROMPT>4876h, 3539m, 22200e, 21648w cexkdb-\27[4z</PROMPT>", "\27[4z<PROMPT>4876h, 3539m, 22200e, 21648w cexkdb-\27[4z</PROMPT>"},
+      }
+      for _, seq in ipairs(sequences) do
+        local actualResult = ansi2decho(seq[1])
+        assert.are.same(seq[2], actualResult)
+      end
+    end)
+
+    it("Should convert xterm256 codes correctly", function()
+      local sequences = {
+        { "\27[38;2;120;134;94m", "<120,134,94:>"},
+        { "\27[48;2;85;250;33m", "<:85,250,33>"},
+        { "\27[38;2;120;134;94;48;2;85;250;33m", "<120,134,94:85,250,33>"},
```

Comment:
As I have commented elsewhere this is not a valid sequence IMHO, `"\27[38;2;120;134;94;" requires another three parameters to be complete before you can move onto another sequence such as "48;2;...m". The missing parameters are "unused{0-255 but ignored};colour-space tolerance{presumably 0-255};colour space {0 or 1}".  The "unused" parameter is for the "black component" for the sequence beginning with "38;4;" or "48;4;" (for CYMK) colour codes presumably so that the other parser can be kept otherwise the same for all the cases where the second parameter is 2 (RGB), 3(CYM) or 4(CYMK)...